server/vulnerabilities/nvd/cpe.go

```diff
@@ -397,12 +397,31 @@ func TranslateSoftwareToCPE(
 		if err != nil {
 			return ctxerr.Wrap(ctx, err, "getting value from iterator")
 		}
-		cpe, err := CPEFromSoftware(db, software, cpeTranslations, reCache)
-		if err != nil {
-			level.Error(logger).Log("software->cpe", "error translating to CPE, skipping...", "err", err)
-			continue
+		var cpe string
+		// Skip software without version to avoid false positives in the CPE
+		// matching process.
+		if software.Version == "" {
+			level.Debug(logger).Log(
+				"msg", "skipping software without version",
+				"software", software.Name,
+				"source", software.Source,
+			)
+		} else {
+			cpe, err = CPEFromSoftware(db, software, cpeTranslations, reCache)
+			if err != nil {
+				level.Error(logger).Log(
+					"msg", "error translating to CPE, skipping",
+					"software", software.Name,
+					"version", software.Version,
+					"source", software.Source,
+					"err", err,
+				)
+				continue
+			}
 		}
 		if cpe == software.GenerateCPE {
```

Comment:
should there be a `continue` in the `software.Version == ""` block?

if `software.Version == ""` then `cpe == ""` and will never(?) match

Reply:
Ah, we really want to continue the flow because in `consumeCPEBuffer` we get rid of entries in `software_cpe` where `cpe == ""` and `software.GenerateCPE != ""` (because of a wrong translation by a previous version of Fleet):
https://github.com/fleetdm/fleet/blob/a5fff556c1bf8d0b8a514076504bd58a3dbd1bb7/server/vulnerabilities/nvd/cpe.go#L326-L331

Reply:
[Add comment to clarify](https://github.com/fleetdm/fleet/pull/14612/commits/07568f073fef442a62b70e8884f72ea8d4c8b137)

Reply:
ah i see, makes sense!